sqlglot/dialects/snowflake.py

```diff
@@ -682,14 +682,23 @@ def _parse_location_property(self) -> exp.LocationProperty:
             return self.expression(exp.LocationProperty, this=self._parse_location_path())
 
         def _parse_file_location(self) -> t.Optional[exp.Expression]:
-            return self._parse_table_parts()
+            # Parse either a subquery or a staged file
+            return (
+                self._parse_select(table=True)
+                if self._match(TokenType.L_PAREN, advance=False)
+                else self._parse_table_parts()
+            )
 
         def _parse_location_path(self) -> exp.Var:
             parts = [self._advance_any(ignore_reserved=True)]
 
             # We avoid consuming a comma token because external tables like @foo and @bar
             # can be joined in a query with a comma separator.
-            while self._is_connected() and not self._match(TokenType.COMMA, advance=False):
+            while (
+                self._is_connected()
+                and not self._match(TokenType.COMMA, advance=False)
+                and not self._match(TokenType.R_PAREN, advance=False)
```

Comment:
was this added to avoid consuming the closing `R_PAREN` of a subquery in `(SELECT ... FROM @foo)`?

Reply:
Right, this combined with the `wrapped` condition in `parse_table_parts()`